### PR TITLE
Move from math/rand to crypto/rand

### PR DESCRIPTION
**Pull request by asitchakraborti-okta**

### CODE REVIEW

**Comment by asitchakraborti-okta** [comment]
math/rand should not be used in any security sensitive context.

File: marshal.go [hunk 1]
```diff
@@ -180,7 +178,17 @@ func marshalOpenssh(pk interface{}, opts *MarshalOptions) ([]byte, error) {
 		blocksize = aes.BlockSize
 	}
 
-	check := mrand.Uint32()
+	// Get a crypto rand in the range [0, uint32-max)
+	randnum, err := rand.Int(rand.Reader, big.NewInt(0x100000000))
```
Comment (evanmoses-okta) [inline marshal.go:182]: Use `math.MaxUint32` instead of this constant

File: marshal.go [hunk 2]
```diff
@@ -180,7 +178,17 @@ func marshalOpenssh(pk interface{}, opts *MarshalOptions) ([]byte, error) {
...
+	if err != nil {
+		return nil, fmt.Errorf("sshkeys: failed to get random number: %w", err)
+	}
+	// Convert it to uint32. As the content was within uint32 limits, nothing should be lost
+	check := uint32(randnum.Uint64())
```
Comment (evanmoses-okta) [inline marshal.go:187]: I think you can trust the library here.  If you really wanna verify, write a generative test using https://pkg.go.dev/testing/quick that'll run this a bunch of times and you can verify stocahstically you'll never get an out-of-range number.

**Review by evanmoses-okta — CHANGES_REQUESTED** [review]

File: marshal.go [hunk 3]
```diff
@@ -180,7 +179,14 @@ func marshalOpenssh(pk interface{}, opts *MarshalOptions) ([]byte, error) {
 		blocksize = aes.BlockSize
 	}
 
-	check := mrand.Uint32()
+	// Get a crypto rand in the range [0, uint32-max)
```
Comment (evanmoses-okta) [inline marshal.go:182]: Comment doesn't align with code.  The range here is `[0, uint32max]`

Comment (asitchakraborti-okta) [reply marshal.go:183]: Actually the rand.Int does the non-inclusive upper bound thing

Comment (evanmoses-okta) [reply marshal.go:183]: right...but you're adding 1

Comment (asitchakraborti-okta) [reply marshal.go:183]: Never mind, let me fix it

**Review by evanmoses-okta — APPROVED** [review]

**Pull request merged**